scripts/deploy-backend.sh

```diff
@@ -1,31 +1,103 @@
-#!/bin/sh
+#!/bin/bash
+
+
+##############################
+# Global Variable Decls 
+##############################
+
 
 # The deployment strategy you wish to employ ( rolling update or setting up a new environment)
 DEPLOY_STRATEGY=${1}
 
 #The application name  defined via the manifest yml for the frontend
-CGHOSTNAME_BACKEND=${2}
+CGAPPNAME_BACKEND=${2}
 
 CF_SPACE=${3}
-
-echo DEPLOY_STRATEGY: "$DEPLOY_STRATEGY"
-echo BACKEND_HOST: "$CGHOSTNAME_BACKEND"
-
 strip() {
     # Usage: strip "string" "pattern"
     printf '%s\n' "${1##$2}"
 }
-#The cloud.gov space defined via environment variable (e.g., "tanf-dev", "tanf-staging")
+# The cloud.gov space defined via environment variable (e.g., "tanf-dev", "tanf-staging")
 env=$(strip $CF_SPACE "tanf-")
 
 
-#Helper method to generate JWT cert and keys for new environment
+echo DEPLOY_STRATEGY: "$DEPLOY_STRATEGY"
+echo BACKEND_HOST: "$CGAPPNAME_BACKEND"
+echo CF_SPACE: "$CF_SPACE"
+echo env: "$env"
+
+
+
+##############################
+# Function Decls
+##############################
+
+
+
+set_cf_envs()
+{
+  var_list=(
+  "ACFTITAN_HOST"
+  "ACFTITAN_KEY"
+  "ACFTITAN_USERNAME"
+  "ACR_VALUES"
+  "AMS_CLIENT_ID"
+  "AMS_CLIENT_SECRET"
+  "AMS_CONFIGURATION_ENDPOINT"
+  "AV_SCAN_URL"
+  "BASE_URL"
+  "CLAMAV_NEEDED"
+  "DJANGO_CONFIGURATION"
+  "DJANGO_SECRET_KEY"
+  "DJANGO_SETTINGS_MODULE"
+  "DJANGO_SU_NAME"
+  "FRONTEND_BASE_URL"
+  "JWT_CERT"
+  "JWT_KEY"
+  "LOGGING_LEVEL"
+  "OIDC_RP_CLIENT_ID"
+  "XMS_CLIENT_ID"
+  "XMS_CLIENT_SECRET")
+
+  for var_name in ${var_list[@]}; do
+
+    # Intentionally not setting variable if empty
+    if [[ -z "${!var_name}" ]]; then
+        echo "WARNING: Empty value for $var_name"
+        continue
+    fi
+    cf_cmd="cf set-env \"$CGAPPNAME_BACKEND\" $var_name \"${!var_name}\""
+    $cf_cmd
+  done
+
+}
+
+set_backend_env_vars()
```

Comment:
I'm not convinced this function is very useful/necessary with that "temporary_deployment" flag vs just calling `set_cf_envs` directly from the other sections of the script. The purpose here is to not blindly loop through each and every var. It feels like the result section could just be moved to after #L70